File: Engine/src/main/java/org/etieskrill/engine/time/LoopPacer.java
package org.etieskrill.engine.time;

//TODO rework interface and document
public interface LoopPacer {
    
    void start();
    
    /**
     * Primary function of a pacer. Must be called exactly once at any point in the target loop.
     */
    void nextFrame();
    
    double getDeltaTimeSeconds();
    double getSecondsElapsedTotal();

    void pauseTimer();
    void resumeTimer();
    boolean isPaused();
    void resetTimer();
    double getTime();

    double getAverageFPS();

    long getTotalFramesElapsed();
    long getFramesElapsed();
    void resetFrameCounter();

    double getTargetDeltaTime();
    void setTargetDeltaTime(double deltaSeconds);
    
}

File: Engine/src/main/java/org/etieskrill/engine/time/LoopPacer.kt
package org.etieskrill.engine.time

//TODO rework interface and document
interface LoopPacer {

    fun start()

    /**
     * Primary function of a pacer. Must be called exactly once at any point in the target loop.
     */
    fun nextFrame()

    fun getDeltaTimeSeconds(): Double
    fun getSecondsElapsedTotal(): Double

    fun pauseTimer()
    fun resumeTimer()
    fun isPaused(): Boolean
    fun resetTimer()
    fun getTime(): Double

    fun getAverageFPS(): Double

    fun getTotalFramesElapsed(): Long
    fun getFramesElapsed(): Long
    fun resetFrameCounter()

    fun getTargetDeltaTime(): Double
    fun setTargetDeltaTime(deltaSeconds: Double)

}
